sky/cli.py

```diff
@@ -1913,28 +1928,67 @@ def _down_or_stop_clusters(
         # Make sure the reserved clusters are explicitly specified without other
         # normal clusters and purge is True.
         if len(reserved_clusters) > 0:
+            if len(names) != 0:
+                names_str = ', '.join(map(repr, names))
+                raise click.UsageError(
+                    f'{operation} reserved cluster(s) '
+                    f'{reserved_clusters_str} with multiple other cluster(s) '
+                    f'{names_str} is not supported.\n'
+                    f'Please omit the reserved cluster(s) {reserved_clusters}.')
             if not down:
                 raise click.UsageError(
                     f'{operation} reserved cluster(s) '
                     f'{reserved_clusters_str} is not supported.')
             else:
-                click.secho(
-                    'WARNING: Tearing down a SkyPilot reserved cluster. If '
-                    'any related job had not finished, resources leakage '
-                    'could happen.',
-                    fg='yellow')
+                # TODO(zhwu): We can only have one reserved cluster (spot
```

Comment:
This func has become too long. Possible to refactor out this block? https://google.github.io/styleguide/pyguide.html#318-function-length